### DISCUSSION TITLE
What kind of environment does `vhs record` put us in?

### DISCUSSION THREAD

**Discussion by suguruwataru** [charmbracelet/vhs#536] [Q&A]
I was trying to record a plugin I wrote for neovim, which includes letting neovim running an external command. In the `vhs record` shell, neovim says it cannot execute the command. There are many other strange things. `which ls` prints `which: no ls in ((null))`. Environment variable `PATH` is set to something different than what I usually have. Actually almost all environmental variables are gone. I once experienced one program not being able to access a Unix domain socket, but can't seem to reproduce this. I thought this was just starting a new shell and proxying  its input and output, apparently it's more than that, so what kind of environment is this?

**Comment by pronebird** [comment]
I'd like to know too.
